src/yamlfix/adapters.py

```diff
@@ -673,7 +673,7 @@ def _replace_whitelines(match: Match[str], n_whitelines: int) -> str:
 
     def _fix_section_whitelines(self, source_code: str) -> str:
         re_section = "\n*(^#.*\n)*\n*^[^ ].*:\n(\n|(^  .*))+\n*"
-        re_beginning_section = f"(?P<b>---\n{re_section})"
+        re_beginning_section = f"(?P<b>(?:---\n|\\A){re_section})"
```

Comment:
I've never heard of the `\A` section beginning? Is this an official thing?

Reply:
Fair point. It was new to me too. I've linked to the Python regex syntax docs.

Reply:
Lol, I thought that the `\A` was a yaml thing not a regexp thing (facepalm). LGTM